src/target/arc_ocd.c

```diff
@@ -61,11 +61,26 @@ int arc_ocd_poll(struct target *target)
 			target->state = TARGET_RUNNING;
 		}
 	} else {
-		if ((target->state == TARGET_RUNNING) ||
-			(target->state == TARGET_RESET)) {
+		/* In some cases JTAG status register indicates that
+		*  processor is in halt mode, but processor is still running.
+		*  We check halt bit of AUX STATUS32 register for setting correct state
+		*/
+		if (target->state == TARGET_RUNNING) {
```

Comment:
I think you can leave

```c
if ((target->state == TARGET_RUNNING) ||
    (target->state == TARGET_RESET)) {
```

as it was. Checking STATUS32 will not harm after reset and will simplify your code - one less branch.